Description: do not crash if an inappropriately long argument is used for -D.
Author: Kees Cook <kees@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/net-tools/+bug/515010
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=569509

--- a/arp.c
+++ b/arp.c
@@ -243,7 +243,8 @@
     struct ifreq ifr;
     struct hwtype *xhw;
 
-    strcpy(ifr.ifr_name, ifname);
+    strncpy(ifr.ifr_name, ifname, sizeof(ifr.ifr_name));
+    ifr.ifr_name[IFNAMSIZ-1]='\0';
     if (ioctl(sockfd, SIOCGIFHWADDR, &ifr) < 0) {
 	fprintf(stderr, _("arp: cant get HW-Address for `%s': %s.\n"), ifname, strerror(errno));
 	return (-1);
